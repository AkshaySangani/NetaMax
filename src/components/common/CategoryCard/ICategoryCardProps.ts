import { ICategory } from "dataflows/Category/ICategory";

export interface ICategoryCardProps {
  /**
   * The categories to display
   * @type {ICategory}
   */
  category: ICategory;

  /**
   * On category click action.
   * @type {(category: ICategory) => void}
   */
  onCategoryClick: (category: ICategory) => void;
}
