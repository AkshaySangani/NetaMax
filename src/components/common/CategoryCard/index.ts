export type { ICategoryCardProps } from "./ICategoryCardProps";
export { CategoryCard } from "./CategoryCard";
