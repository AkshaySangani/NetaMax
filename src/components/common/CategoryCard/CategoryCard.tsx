import React, { ReactElement } from "react";

import { Box, Image } from "@chakra-ui/react";

import { ICategoryCardProps } from "./ICategoryCardProps";

/**
 * The CategoryBox component.
 * @param {ICategoryCardProps} props the component props.
 * @returns {ReactElement} the category card component.
 */
export const CategoryCard = (props: ICategoryCardProps): ReactElement => {
  const { category, onCategoryClick } = props;

  return (
    <Box maxW="sm" height="100%" overflow="hidden">
      <Image
        src={category.seoFilename}
        draggable="false"
        alt={category.name}
        height="100%"
        minW={{ base: 159, md: "160px" }}
        onClick={() => onCategoryClick(category)}
        fallbackSrc="/assets/images/noimg.png"
      />
    </Box>
  );
};
