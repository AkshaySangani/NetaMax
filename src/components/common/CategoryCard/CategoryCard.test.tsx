import { render, cleanup, fireEvent } from "@testing-library/react";
import { CategoryCard } from "./CategoryCard";
import { ICategoryCardProps } from "./ICategoryCardProps";

describe("CategoryCard Component", () => {
  const onCategoryClick = jest.fn();
  let mock: ICategoryCardProps;

  beforeEach(() => {
    mock = {
      category: {
        id: "1978",
        name: "Huevo blanco San Juan 12 pzas",
        seoFilename: "https://nopcommerce-assets.s3.us-east-2.amazonaws.com/0002120_0.jpeg",
        createdOnUtc: new Date(),
        updatedOnUtc: new Date(),
      },
      onCategoryClick,
    };
  });

  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
  });

  test("CategoryCard should be render correctly", () => {
    const { container } = render(<CategoryCard {...mock} />);
    expect(container).toMatchSnapshot();
  });

  test("Alt text of image should not be null", () => {
    const { getByAltText } = render(<CategoryCard {...mock} />);
    expect(getByAltText("Huevo blanco San Juan 12 pzas")).toBeInTheDocument();
  });

  test("CategoryCard should be clickable", () => {
    const { getByAltText } = render(<CategoryCard {...mock} />);
    fireEvent.click(getByAltText("Huevo blanco San Juan 12 pzas"));
    expect(onCategoryClick).toBeCalledTimes(1);
    expect(onCategoryClick).toHaveBeenCalledWith(mock.category);
  });
});
