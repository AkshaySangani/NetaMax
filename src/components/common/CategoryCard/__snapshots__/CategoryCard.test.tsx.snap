// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryCard Component CategoryCard should be render correctly 1`] = `
<div>
  <div
    class="css-n29tsx"
  >
    <img
      alt="Huevo blanco San Juan 12 pzas"
      class="chakra-image__placeholder css-853n4v"
      draggable="false"
      src="/assets/images/noimg.png"
    />
  </div>
</div>
`;
